flag.go

```diff
@@ -825,16 +834,20 @@ func (f *FlagSet) parseSingleShortArg(shorthands string, args []string) (outShor
 		err = f.failf("flag needs an argument: %q in -%s", c, shorthands)
 		return
 	}
-	err = f.setFlag(flag, value, shorthands)
-	return
+
+	origVal := value
+	if defVal {
+		origVal = ""
+	}
+	return outShorts, outArgs, fn(flag, value, shorthands, "-"+shorthands[:1], origVal)
```

Comment:
Two doubts here. doesn't set the value to flag, and shorthands may not be the real origArg.  People maybe fall into the trap.